src/SFA.DAS.EAS.Web.IntegrationTests/TestUtils/ApiIntegrationTester.cs

```diff
@@ -0,0 +1,196 @@
+        {
+            using (var tester = new ApiIntegrationTester() )
+            {
+                return await tester.InvokeGetAsync(call);
+            }
+        }
+
+        public void BeginTests()
+        {
+            TestServer = TestServer.Create(InitialiseHost);
+        }
+
+        public void Dispose()
+        {
+            if (IsTestServerStarted)
+            {
+                EndTests();
+            }
+        }
+
+        public void EndTests()
+        {
+            TestServer?.Dispose();
+            TestServer = null;
+        }
+
+        public Task InvokeGetAsync(CallRequirements call)
+        {
+            return GetResponseAsync(call);
+        }
+
+        public async Task<TResult> InvokeGetAsync<TResult>(CallRequirements<TResult> call)
+        {
+            var result = await GetResponseAsync(call);
+            call.Result = await FetchCompleteResponse<TResult>(result);
+            return call.Result;
+        }
+
+        private async Task<HttpResponseMessage> GetResponseAsync(CallRequirements call)
+        {
+            EnsureStarted();
+            ClearDownForNewTest();
+            var result = await GetClient(call).GetAsync(call.Uri);
+            CheckCallWasSuccessful(result, call);
+            return result;
+        }
+
+        private HttpClient GetClient(CallRequirements call)
+        {
+            var client = TestServer.HttpClient;
+            client.Timeout = call.TimeOut;
+            return client;
+        }
+
+        private void ClearDownForNewTest()
+        {
+            _exceptionHandler.ClearException();
+            _dependencyResolver.ClearCreationLog();
+        }
+
+        private void EnsureStarted()
+        {
+            if (!IsTestServerStarted)
+            {
+                BeginTests();
+            }
+        }
+
+        private void InitialiseHost(IAppBuilder app)
+        {
+            HttpConfiguration config = new HttpConfiguration();
+            WebApiConfig.Register(config);
+            CustomiseIoC(config);
+            app.UseWebApi(config);
+        }
+
+        private void CustomiseIoC(HttpConfiguration configuration)
+        {
+            var existingExceptionHandler = (IExceptionHandler) configuration.Services.GetService(typeof(IExceptionHandler));
+            _exceptionHandler = new IntegrationTestExceptionHandler(existingExceptionHandler);
+
+            configuration.Services.Replace(typeof(IAssembliesResolver), new TestWebApiResolver<AccountLegalEntitiesController>());
+            configuration.Services.Replace(typeof(IExceptionHandler), _exceptionHandler);
+
+            StructuremapMvc.Start();
+            StructuremapWebApi.Start();
+            var container = ((StructureMapWebApiDependencyResolver)GlobalConfiguration.Configuration.DependencyResolver).Container;
+            var requestContextMock = new Mock<IRequestContext>();
+            container.Inject(requestContextMock.Object);
```

Comment:
Just a thought but could a StructureMap registry just for the `SFA.DAS.EAS.Account.API.IntegrationTests` project be used for any overrides?